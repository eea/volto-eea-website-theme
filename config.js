import navigationSettings from './src/reducers/navigation/navigation';

export default function applyConfig(config) {
  config.addonReducers = {
    ...config.addonReducers,
    navigationSettings,
  };
  return config;
}