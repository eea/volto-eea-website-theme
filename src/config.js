import biseLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/bise.svg';
import energyLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/energy.svg';
import insituLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/insitu.svg';
import freshwaterLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/freshwater.svg';
import fiseLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/fise.svg';
import ccaLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/cca.svg';
import copernicusLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/copernicus.svg';
import industryLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/industry.svg';
import marineLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/marine.svg';
import eionetLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/eionet.svg';
import eeaLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/images/Header/eea-logo.svg';
import eeaWhiteLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/eea-white.svg';
import climateLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/climate-health.svg';

// TODO: to be consolidated with headerLinks
// Footer.jsx config options
export const footerOpts = {
  header: 'Thematic information platforms',
  logosHeader: 'Managed by',
  contactHeader: 'Contact Us',
  actions: [
    {
      title: 'Privacy',
      link: '/privacy',
    },
    {
      link: '/sitemap',
      title: 'Sitemap',
    },
    {
      link: '/login',
      title: 'CMS Login',
    },
  ],
  copyright: [
    {
      link: '/copyright',
      site: 'EEA',
    },
  ],
  sites: [
    {
      link: 'https://water.europa.eu/marine',
      src: marineLogo,
      alt: 'WISE marine',
    },
    {
      link: 'https://water.europa.eu/freshwater',
      src: freshwaterLogo,
      alt: 'WISE freshwater',
    },
    {
      link: 'https://biodiversity.europa.eu/',
      src: biseLogo,
      alt: 'Biodiversity',
    },
    {
      link: 'https://forest.eea.europa.eu/',
      src: fiseLogo,
      alt: 'Forest information system for europe',
    },
    {
      link: 'https://climate-adapt.eea.europa.eu/observatory',
      src: climateLogo,
      alt: 'European Climate and Health Observatory',
    },
    {
      link: 'https://climate-adapt.eea.europa.eu/',
      src: ccaLogo,
      alt: 'Climate adapt',
    },
    {
      link: 'https://industry.eea.europa.eu/',
      src: industryLogo,
      alt: 'European industrial emissions portal',
    },
    {
      link: 'https://climate-energy.eea.europa.eu/',
      src: energyLogo,
      alt: 'Climate and energy in the EU',
    },
    {
      link: 'https://land.copernicus.eu/',
      src: copernicusLogo,
      alt: 'Copernicus land monitoring service',
    },
    {
      link: 'https://insitu.copernicus.eu/',
      src: insituLogo,
      alt: 'Copernicus in situ',
    },
  ],
  managedBy: [
    {
      link: 'https://www.eea.europa.eu/',
      src: eeaWhiteLogo,
      alt: 'EEA Logo',
      className: 'site logo',
      columnSize: {
        mobile: 6,
        tablet: 12,
        computer: 4,
      },
    },
    {
      link: 'https://www.eionet.europa.eu/',
      src: eionetLogo,
      alt: 'EIONET Logo',
      className: 'eionet logo',
      columnSize: {
        mobile: 6,
        tablet: 12,
        computer: 4,
      },
    },
  ],
  social: [
    {
      name: 'twitter',
      icon: 'ri-twitter-fill',
      link: 'https://twitter.com/euenvironment',
    },
    {
      name: 'facebook',
      icon: 'ri-facebook-box-fill',
      link: 'https://www.facebook.com/European.Environment.Agency',
    },
    {
      name: 'linkedin',
      icon: 'ri-linkedin-fill',
      link: 'https://www.linkedin.com/company/european-environment-agency',
    },
    {
      name: 'youtube',
      icon: 'ri-youtube-fill',
      link: 'https://www.youtube.com/user/EEAvideos',
    },
    {
      name: 'instagram',
      icon: 'ri-instagram-fill',
      link: 'https://www.instagram.com/ourplanet_eu',
    },
    {
      name: 'rss',
      icon: 'ri-rss-fill',
      link: '/subscription/news-feeds',
    },
  ],
  contacts: [
    {
      icon: 'comment outline',
      text: 'About us',
      link: '/about',
      children: [
        {
          link: '/faq',
          text: 'FAQs',
        },
        {
          link: '/careers',
          text: 'Careers',
        },
      ],
    },
    {
      icon: 'comment outline',
      text: 'Contact us',
      link: '/contact-us',
    },
    {
      icon: 'envelope outline',
      text: 'Sign up to our newsletter',
      link: '/newsletter',
    },
  ],
  address: 'Kongens Nytorv 6 1050 Copenhagen K (+45) 33 36 71 00',
};

// Header.jsx config options
export const headerOpts = {
  logo: eeaLogo,
  logoWhite: eeaWhiteLogo,
  partnerLinks: {
    title: 'Environmental information systems',
    links: [
      {
        title: 'Biodiversity Information System for Europe',
        href: 'https://biodiversity.europa.eu/',
      },
      {
        title: 'Climate Adaptation Platform',
        href: 'https://climate-adapt.eea.europa.eu/',
      },
      {
        title: 'Copernicus in situ component',
        href: 'https://insitu.copernicus.eu/',
      },
      {
        title: 'European Industrial Emissions Portal',
        href: 'https://industry.eea.europa.eu/',
      },
      {
        title: 'Forest Information System for Europe',
        href: 'https://forest.eea.europa.eu/',
      },
      {
        title: 'Information Platform for Chemical Monitoring',
        href: 'https://ipchem.jrc.ec.europa.eu/RDSIdiscovery/ipchem/index.html',
      },
      {
        title: 'Marine Water Information System for Europe',
        href: 'https://water.europa.eu/marine',
      },
      {
        title: 'Fresh Water Information System for Europe',
        href: 'https://water.europa.eu/freshwater',
      },
    ],
  },
};

export const languages = [
  { name: 'Български', code: 'bg' },
  { name: 'čeština', code: 'cs' },
  { name: 'Hrvatski', code: 'hr' },
  { name: 'dansk', code: 'da' },
  { name: 'Nederlands', code: 'nl' },
  { name: 'ελληνικά', code: 'el' },
  { name: 'English', code: 'en' },
  { name: 'eesti', code: 'et' },
  { name: 'Suomi', code: 'fi' },
  { name: 'Français', code: 'fr' },
  { name: 'Deutsch', code: 'de' },
  { name: 'magyar', code: 'hu' },
  { name: 'Íslenska', code: 'is' },
  { name: 'italiano', code: 'it' },
  { name: 'Latviešu', code: 'lv' },
  { name: 'lietuvių', code: 'lt' },
  { name: 'Malti', code: 'mt' },
  { name: 'Norsk', code: 'no' },
  { name: 'polski', code: 'pl' },
  { name: 'Português', code: 'pt' },
  { name: 'Română', code: 'ro' },
  { name: 'slovenčina', code: 'sk' },
  { name: 'Slovenščina', code: 'sl' },
  { name: 'Español', code: 'es' },
  { name: 'Svenska', code: 'sv' },
  { name: 'Türkçe', code: 'tr' },
];

export const defaultLanguage = 'en';

export const websiteTitle = 'Site';
export const organisationName = 'European Environment Agency';
export const logoTargetUrl = '/';

export const headerSearchBox = [
  {
    isDefault: true,
    path: '/en/advanced-search',
    placeholder: 'Search...',
  },
  {
    path: '/en/datahub',
    placeholder: 'Search Datahub...',
    description:
      'Looking for more information? Try searching the full EEA website content',
    buttonTitle: 'Go to full site search',
  },
];

export const colors = [
  // Primary & shades
  '#007B6C',
  '#005248',
  '#007B6C',
  '#00A390',
  '#00CCB4',
  '#00F5D8',
  '#85FFF1',
  '#C8FFF8',
  // Secondary & shades
  '#004B7F',
  '#003052',
  '#004B7F',
  '#006BB8',
  '#0083E0',
  '#0A99FF',
  '#47B3FF',
  '#A0D7FF',
  // Tertiary & shades
  '#2E3E4C',
  '#3D5265',
  '#4C677F',
  '#6989A5',
  '#8EA6C2',
  '#ACCAE5',
  '#DAE8F4',
  // Black & White shades
  '#000000',
  '#323232',
  '#67696b',
  '#BCBEC0',
  '#E6E7E8',
  '#F9F9F9',
  '#FFFFFF',
];
