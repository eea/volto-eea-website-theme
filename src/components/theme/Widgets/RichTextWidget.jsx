/**
 * A Slate widget that uses internal JSON representation as its value
 *
 */

import React from 'react';
import isUndefined from 'lodash/isUndefined';
import isString from 'lodash/isString';
import { FormFieldWrapper } from '@plone/volto/components';
import SlateEditor from '@plone/volto-slate/editor/SlateEditor';

import {
  createEmptyParagraph,
  createParagraph,
} from '@plone/volto-slate/utils/blocks';
import { handleKeyDetached } from '@plone/volto-slate/blocks/Text/keyboard';
import config from '@plone/volto/registry';

//import './style.css';

const getValue = (value) => {
  if (isUndefined(value) || !isUndefined(value?.data)) {
    return [createEmptyParagraph()];
  }
  // Previously this was a text field
  if (isString(value)) {
    return [createParagraph(value)];
  }
  return value;
};

export const RichTextWidget = (props) => {
  const {
    id,
    onChange,
    value,
    focus,
    className,
    block,
    placeholder,
    properties,
    readOnly = false,
  } = props;
  const [selected, setSelected] = React.useState(focus);
  const { slateWidgetExtensions } = config.settings.slate;

  return (
    <FormFieldWrapper {...props} draggable={false} className="slate_wysiwyg">
      <div
        className="slate_wysiwyg_box"
        role="textbox"
        tabIndex="-1"
        style={{ boxSizing: 'initial' }}
        onClick={() => {
          setSelected(true);
        }}
        onKeyDown={() => {}}
      >
        <SlateEditor
          className={className}
          readOnly={readOnly}
          id={id}
          name={id}
          value={getValue(value)}
          onChange={(newValue) => {
            onChange(id, newValue);
          }}
          block={block}
          selected={selected}
          properties={properties}
          placeholder={placeholder}
          extensions={slateWidgetExtensions}
          onKeyDown={handleKeyDetached}
          editableProps={{ 'aria-multiline': 'true' }}
        />
      </div>
    </FormFieldWrapper>
  );
};
