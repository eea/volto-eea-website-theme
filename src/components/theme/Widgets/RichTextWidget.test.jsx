import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import config from '@plone/volto/registry';
import { RichTextWidget } from './RichTextWidget';

jest.mock('@plone/volto-slate/editor/SlateEditor', () => {
  return {
    __esModule: true,
    default: ({ value, onChange, selected = false }) => (
      <div data-testid="slate-editor">
        <span>Mocked Slate Editor</span>
        <span>Selected: {selected.toString()}</span>
        <span>Value: {JSON.stringify(value)}</span>
      </div>
    ),
  };
});

const mockStore = configureStore();
beforeAll(() => {
  config.settings = {
    slate: {
      slateWidgetExtensions: [],
    },
  };
});

const store = mockStore({
  intl: {
    locale: 'en-gb',
    messages: {},
  },
});

describe('RichTextWidget', () => {
  const defaultProps = {
    id: 'test-rich-text',
    name: 'test-rich-text',
    value: '',
    onChange: jest.fn(),
    selected: false,
  };
  it('renders an empty rich text widget component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <RichTextWidget {...defaultProps} />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a rich text widget component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <RichTextWidget
          className="metadata"
          value={{ data: '<b>Foo bar</b>' }}
        />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a rich text view widget component with children', () => {
    const component = renderer.create(
      <Provider store={store}>
        <RichTextWidget className="metadata" value={{ data: '<b>Foo bar</b>' }}>
          {(child) => <strong>{child}</strong>}
        </RichTextWidget>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('renders with mocked SlateEditor', () => {
    const component = renderer.create(
      <Provider store={store}>
        <RichTextWidget {...defaultProps} />
      </Provider>,
    );

    const componentInstance = component.root;
    const element = componentInstance.find(
      (node) => node.props['data-testid'] === 'slate-editor',
    );

    expect(element).toBeDefined();
    expect(element.props['data-testid']).toBe('slate-editor');
  });
});
