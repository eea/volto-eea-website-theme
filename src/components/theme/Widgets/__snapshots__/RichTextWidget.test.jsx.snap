// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextWidget renders a rich text view widget component with children 1`] = `
<div
  className="inline field slate_wysiwyg field-wrapper-undefined"
>
  <div
    className="ui grid"
  >
    <div
      className="stretched row"
    >
      <div
        className="four wide column"
      >
        <div
          className="wrapper"
        >
          <label
            htmlFor="field-undefined"
            id="fieldset-undefined-field-label-undefined"
          />
        </div>
      </div>
      <div
        className="eight wide column"
      >
        <div
          className="slate_wysiwyg_box"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="textbox"
          style={
            Object {
              "boxSizing": "initial",
            }
          }
          tabIndex="-1"
        >
          <div
            data-testid="slate-editor"
          >
            <span>
              Mocked Slate Editor
            </span>
            <span>
              Selected: 
              false
            </span>
            <span>
              Value: 
              [{"children":[{"text":""}]}]
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RichTextWidget renders a rich text widget component 1`] = `
<div
  className="inline field slate_wysiwyg field-wrapper-undefined"
>
  <div
    className="ui grid"
  >
    <div
      className="stretched row"
    >
      <div
        className="four wide column"
      >
        <div
          className="wrapper"
        >
          <label
            htmlFor="field-undefined"
            id="fieldset-undefined-field-label-undefined"
          />
        </div>
      </div>
      <div
        className="eight wide column"
      >
        <div
          className="slate_wysiwyg_box"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="textbox"
          style={
            Object {
              "boxSizing": "initial",
            }
          }
          tabIndex="-1"
        >
          <div
            data-testid="slate-editor"
          >
            <span>
              Mocked Slate Editor
            </span>
            <span>
              Selected: 
              false
            </span>
            <span>
              Value: 
              [{"children":[{"text":""}]}]
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RichTextWidget renders an empty rich text widget component 1`] = `
<div
  className="inline field slate_wysiwyg field-wrapper-test-rich-text"
>
  <div
    className="ui grid"
  >
    <div
      className="stretched row"
    >
      <div
        className="four wide column"
      >
        <div
          className="wrapper"
        >
          <label
            htmlFor="field-test-rich-text"
            id="fieldset-undefined-field-label-test-rich-text"
          />
        </div>
      </div>
      <div
        className="eight wide column"
      >
        <div
          className="slate_wysiwyg_box"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="textbox"
          style={
            Object {
              "boxSizing": "initial",
            }
          }
          tabIndex="-1"
        >
          <div
            data-testid="slate-editor"
          >
            <span>
              Mocked Slate Editor
            </span>
            <span>
              Selected: 
              false
            </span>
            <span>
              Value: 
              [{"children":[{"text":""}]}]
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
